### PR TITLE
Document keyboard ghosting caveats in Input and InputEvent

### LINKED ISSUE #56423: Document keyboard ghosting issues
### Godot version

3.4.2.stable

### System information

Ubuntu 20.04.3 LTS, Gnome version 3.36.8, X11 windowing system, Dell XPS 13 9360, i7-8550U CPU

### Issue description

While I hold down the up and left arrow keys together, the spacebar stops working (`Input.is_action_pressed` stops returning `true`). The reverse is also true, when I hold down spacebar, the up and left arrow keys don't work together: only one of them works.

This issue doesn't occur:
* For other arrow keys
* For WASD keys
* Something other than spacebar (like LSHIFT)

### Steps to reproduce

1. Run the project using F5 or open the `Level` scene and use F6.
2. Walk around using the arrow keys or WASD.
3. Hold spacebar to change the character's color.
4. Press the up and right arrow keys together and while moving, press spacebar. This will work. The character will change its color while moving diagonally.
5. Press the up and left arrow keys together and do the same thing. This will not work. The character won't change its color.
6. Press the W and A keys together and repeat. This will work.
7. Press the up and left arrow keys together but press LSHIFT instead of spacebar. This will work.
8. Press spacebar and _then_ press the up and left arrow keys together. The character's color will change but it'll only move in either the up or left direction; not diagonally.

### Minimal reproduction project

[spacebar_issue.zip](https://github.com/godotengine/godot/files/7798651/spacebar_issue.zip)

### PR DESCRIPTION
**Pull request by Calinou**
This also adds recommendations for choosing between `is_physical_key_pressed()` and `is_key_pressed()`.

This closes https://github.com/godotengine/godot/issues/56423.

### CODE REVIEW

**Comment by mhilbrunner** [comment]
I feel like in contrast to the other PR (https://github.com/godotengine/godot-docs/pull/5496), this is a bit lacking in actually actionable advice. Notably:
- That this is a hardware-side, not a Godot-specific issue (mostly on lower end hardware)
- That it should be anticipated and though of when designing the game's keyboard layout

However, this would make these even worider. What do you think about referencing the doc from the above PR/adding a link there?

**Comment by Calinou** [comment]
> What do you think about referencing the doc from the above PR/adding a link there?

Done :slightly_smiling_face:

**Review by mhilbrunner — APPROVED** [review]

**Review by akien-mga — APPROVED** [review]
Still feels a bit wordy to document something which isn't a Godot issue in do many locations, but too much information is better than no information, especially for users who are not familiar with ghosting.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.3.
